### ISSUE TITLE
Incorrect version of Spring Cloud GCP for Spring Boot 2.1

### ISSUE THREAD

**Issue by meltsufin** [spring-io/initializr#742]
Ever since `Greenwich.M1` Spring Cloud GCP is part of the Spring Cloud BOM, and doesn't need its own BOM to be included in dependency management. In addition, there's a problem with the generated `pom.xml` currently, where `1.0.0.RELEASE` is always selected for `spring-cloud-gcp`.

If I select Spring Boot 2.1, I get 
```xml
<spring-cloud-gcp.version>1.0.0.RELEASE</spring-cloud-gcp.version>
<spring-cloud.version>Greenwich.M1</spring-cloud.version>
```
which is clearly wrong because, the only version of Spring Cloud GCP that is compatible with SB 2.1 is `1.1.0.M1`.

If someone is using SB 2.0, however, the current behavior is fine.

cc: @saturnism @snicoll

**Comment by snicoll** [comment]
Update to the metadata can't really happen automatically and someone should trigger it via an issue or a PR. I saw that Spring Cloud GCP joined the released train but it didn't register at the time.

There is no way to add a bom according to a Spring Boot version so I am afraid that we can't really fix the situation right now (other than upgrading the GCP version so that it matches the version that Greenwich brings.

**Comment by meltsufin** [comment]
Thanks @snicoll! I'm ok with the spring-cloud-gcp BOM still being there. In a way it's actually better, because it allows us to release more version outside of the release train, and have the user easily switch to those if needed. I think you can close this issue now.

**Comment by snicoll** [comment]
>  I think you can close this issue now.

I disagree. If you've joined the release train, I'd like to use a consistent experience with an harmonized dependency management. This also reduces the burden on our side (upgrading to the release train rather than having to track for sub-projects). This helped me find out about https://github.com/spring-io/initializr/issues/743
